src/base/base.scss

```diff
@@ -76,10 +76,43 @@ button {
 }
 
 details {
-  summary { cursor: pointer; }
+  summary {
+    cursor: pointer;
+  }
 
   &:not([open]) {
     // Set details content hidden by default for browsers that don't do this
-    > *:not(summary) { display: none !important; }
+    > *:not(summary) {
+      display: none !important;
+    }
+  }
+}
+
+// global focus styles
+
+// fallback :focus state
+:focus,
+.focus {
+  &:is(a, button, input, textarea, select) {
+    @include focusOutline;
+  }
+}
+
+// remove fallback :focus if :focus-visible is supported
+:focus:is(a, button, input, textarea, select):not(:focus-visible) {
+  outline: solid 1px transparent;
+}
+
+// default :focus-visibile state
+:focus-visible:is(a, button, input, textarea, select) {
+  @include focusOutline;
+}
+
+// remove offset for form fields
+:focus,
+:focus-visible {
+  &:is(input, textarea, select) {
+    border-color: transparent;
+    outline-offset: 0;
```

Comment:
Should we add `:not([type='radio']):not([type='checkbox'])` to the selector? So that radio buttons and checkboxes keep the 2px gap:

![Screen Shot 2021-11-11 at 17 27 21](https://user-images.githubusercontent.com/378023/141265106-ff5c8d4f-6a53-4c06-a161-e46b0459f7cf.png)
![Screen Shot 2021-11-11 at 17 27 42](https://user-images.githubusercontent.com/378023/141265107-247fb79a-74e2-4e92-9193-e989ac54e799.png)